assets/js/main.js

```diff
@@ -0,0 +1,308 @@
+      
+          <div class="icons">
+          <a onclick="deleteTask(${i})" class="deleteIcon d-block text-danger fs-5" href="#"><i class="fa fa-trash" aria-hidden="true"></i></a>
+          <a onclick="getTaskData(${i})" class="readIcon d-block text-warning fs-5" href="#"><i class="fa fa-id-card" aria-hidden="true"></i></a>
+          <a  class="readIcon d-block text-success fs-5" href="#"><i class="fa fa-check-circle" aria-hidden="true"></i></a>
+          
+          </div>
+      
+          </div>
+      
+          `;
+    }
+  }
+  tasksDataDone.innerHTML = result;
+}
+
+// clear data after add task
+function clear() {
+  taskName.value = " ";
+  assignee.value = " ";
+}
+
+// delete task when click on delete icon
+function deleteTask(index) {
+  // sweet alert
+  Swal.fire({
+    title: "Are you sure?",
+    text: "You won't be able to revert this!",
+    icon: "warning",
+    showCancelButton: true,
+    confirmButtonColor: "#3085d6",
+    cancelButtonColor: "#d33",
+    confirmButtonText: "Yes, delete it!",
+  }).then((result) => {
+    if (result.isConfirmed) {
+      tasksArray.splice(index, 1);
+      localStorage.setItem("tasksList", JSON.stringify(tasksArray));
+      displayData();
+      counter();
+
+      Swal.fire("Deleted!", "Your task has been deleted.", "success");
+    }
+  });
+}
+
+// To delete all tasks
+deleteAllButton.onclick = function () {
+  // sweet alert
+  Swal.fire({
+    title: "Are you sure?",
+    text: "You won't be able to revert this!",
+    icon: "warning",
+    showCancelButton: true,
+    confirmButtonColor: "#3085d6",
+    cancelButtonColor: "#d33",
+    confirmButtonText: "Yes, delete all !",
+  }).then((result) => {
+    if (result.isConfirmed) {
+      tasksArray.splice(0);
+      localStorage.setItem("tasksList", JSON.stringify(tasksArray));
+      displayData();
+      counter();
+      Swal.fire("Deleted!", "All task have been deleted.", "success");
+    }
+  });
+};
+
+// Search function
+function search(e) {
+  var result = "";
+  for (let i = 0; i < tasksArray.length; i++) {
+    if (tasksArray[i].name.toLowerCase().includes(e.value.toLowerCase())) {
+      result += `
+          <div class="taskDiv d-flex justify-content-between">
+      
+          <div class="taskAndAssignee">
+          <p class="taskNamePara pb-2">${tasksArray[i].name}</p>
+          <p class="assigneePara">${tasksArray[i].assignee}</p>
+          </div>
+      
+          <div class="icons">
+          <a onclick="deleteTask(${i})" class="deleteIcon d-block text-danger fs-5" href="#"><i class="fa fa-trash" aria-hidden="true"></i></a>
+          <a onclick="getTaskData(${i})" class="readIcon d-block text-warning fs-5" href="#"><i class="fa fa-id-card" aria-hidden="true"></i></a>
+          <a class="readIcon d-block text-success fs-5" href="#"><i class="fa fa-check-circle" aria-hidden="true"></i></a>
+          </div>
+      
+          </div>
+      
+          `;
+    }
+  }
+
+  debounceUpdate(result);
```

Comment:
i dont think the debounce here is really useful. The purpose of using debounce is to enhance performance and you are not really doing that here.

you should call debounce earlier in the process before searching the array itself. As we discussed in the call, Imagine that you are searching through an array that contains thousands of records, or you are calling an API to apply this search. With your usage of using debounce, you will keep doing the search on every change to your search input, which is not good.